CookingCurator/Controllers/HomeController.cs

```diff
@@ -140,5 +142,92 @@ public ActionResult AcceptWaiver(UserAcceptWaiverViewModel user)
                 return RedirectToAction("Index");
             }
         }
+
+        public ActionResult Forgot()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        public ActionResult Forgot(RecoverViewModel recoverModel)
+        {
+
+            var u = m.GetUserByEmail(recoverModel.userEmail);
+
+            //Vague message to prevent guessing emails
+            if(u == null)
+            {
+                ModelState.AddModelError("", "An email has been sent to the specified email, provided it was used to make a Cooking Curator Account");
+                return View();
+            }
+
+            recoverModel.banUser = u.banUser;
+            recoverModel.email_Verified = u.email_Verified;
+            recoverModel.userName = u.userName;
+
+            bool error = m.RecoverUser(recoverModel);
+            if (!error)
+            {
+                ModelState.AddModelError("", "An email has been sent to the specified email, provided it was used to make a Cooking Curator Account");
+                u.GUID = recoverModel.GUID;
+                return View();
+            }
+            else
+            {
+                //Account is banned or email was never verified
+                ModelState.AddModelError("", "The account beloning to the specified email address can not be recovered.");
+                return View();
+            }
+
+        }
+
+        [Route("reset/{id}")]
+        public ActionResult Reset(String id)
+        {
+            return View();
+        }
+
+        [HttpPost]
+        [Route("reset/{id}")]
+        public ActionResult Reset(RecoverViewModel resetModel, String id)
+        {
+
+            //Take the GUID from the URL
+            resetModel.GUID = id;
+
+            //Passwords must match
+            if(resetModel.confirmPassword != resetModel.password)
+            {
+                ModelState.AddModelError("", "Error: The passwords you entered do not match.");
+                return View();
+            }
+
+            //Check for special characters
+            Regex r = new Regex("^[a-zA-Z0-9_]*$");
```

Comment:
I don't know if we should have validation logic in the controller. I think we should have it in the view model or the manager to have cleaner code